Common/src/Storage/TaskLifeCycleHelper.cs

```diff
@@ -469,4 +469,31 @@ await taskTable.FinalizeTaskCreation(messages.Select(task => task.TaskId)
                                          cancellationToken)
                    .ConfigureAwait(false);
   }
+
+  public static async Task<SessionData> ResumeAsync(ITaskTable        taskTable,
+                                                    ISessionTable     sessionTable,
+                                                    IPushQueueStorage pushQueueStorage,
+                                                    string            sessionId,
+                                                    CancellationToken cancellationToken = default)
+  {
+    await foreach (var grouping in taskTable.FindTasksAsync(data => data.SessionId == sessionId && data.Status == TaskStatus.Submitted,
+                                                            data => new MessageData(data.TaskId,
+                                                                                    data.SessionId,
+                                                                                    data.Options),
+                                                            cancellationToken)
+                                            .GroupBy(msg => (msg.Options.PartitionId, msg.Options.Priority))
+                                            .WithCancellation(cancellationToken)
+                                            .ConfigureAwait(false))
+    {
+      await pushQueueStorage.PushMessagesAsync(await grouping.ToListAsync(cancellationToken)
+                                                             .ConfigureAwait(false),
+                                               grouping.Key.PartitionId,
+                                               cancellationToken)
+                            .ConfigureAwait(false);
+    }
```

Comment:
I think this code already exist somewhere. Maybe in the agent, or in FinalizeTaskCreation

Reply:
Yeah but not quite. It is the function above. It also FinalizeTasKCreation which we do not want to do here

Reply:
It is also async whereas the other is not